controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -650,11 +651,24 @@ func (r *BareMetalHostReconciler) preprovImageAvailable(info *reconcileInfo, ima
 	return false, nil
 }
 
+func hasArchAnnotation(host *metal3v1alpha1.BareMetalHost) (bool, string) {
+	annotations := host.GetAnnotations()
+	if annotations != nil {
+		if cpuArchitecture, ok := annotations[cpuArchitectureAnnotation]; ok && cpuArchitecture != "" {
+			return true, cpuArchitecture
+		}
+	}
+	return false, ""
+}
+
 func getHostArchitecture(host *metal3v1alpha1.BareMetalHost) string {
 	if host.Status.HardwareDetails != nil &&
 		host.Status.HardwareDetails.CPU.Arch != "" {
 		return host.Status.HardwareDetails.CPU.Arch
 	}
+	if has, architecture := hasArchAnnotation(host); has {
+		return architecture
+	}
```

Comment:
Even if the architecture of the nodes is different, this function can also get the architecture of the current host, so I think it can meet your needs without the code you add.

Reply:
host.Status.HardwareDetails.CPU.Arch comes from the hardware information collected by inspect. My idea is to know the architecture of bmh before inspecting, so as to specify the image of the corresponding architecture for ironic @zhouhao3

Reply:
> My idea is to know the architecture of bmh before inspecting, so as to specify the image of the corresponding architecture for ironic.

I don't understand why you would do this.
Currently this function is only called when preparing the provision image. So even if you want to meet your ideas, you should not change it here.

Reply:
Sorry, I have some misunderstandings here, can you provide a contact method to continue the discussion with you, thank you @zhouhao3